### ISSUE TITLE
Issue: duplicate changelog notes being generated when git < 2.13

### ISSUE THREAD

**Issue by matthias-ccri** [microsoft/beachball#516]
Hi there, we were seeing beachball generate duplicate changelog notes when releasing. We figured out the issue, so I want to share what the issue is for others who might be seeing the same thing.

Background: we were using this docker image: `node:12.13.1-stretch`. Turns out it is an issue with the version of git that the
image uses.

When Beachball runs the publish command it starts by updating the CHANGELOG files, deleting the change files, and then calling npm publish to push the package to the repo. Then it moves on to pushing the updates to git and it calls `revertLocalChanges` from the 'workspace-tools' here: https://github.com/microsoft/beachball/blob/ee51836e3e95d82aa48a6699e469ea4ad1df6c03/src/publish/bumpAndPush.ts#L22
Then it redoes the bump changes and merges main before pushing to origin. The `revertLocalChanges` method (https://github.com/microsoft/workspace-tools/blob/master/src/git.ts#L323) uses `git stash push` to stash and drop the CHANGELOG changes from the publish step. **The problem is that 'push' wasn't added as an option to git stash until 2.13** and node:12.13.1-stretch comes with git 2.11.0 so running the git command in the revert function gives:

```
root@1c76e839f06b:/opt/repository# git stash push -u -m beachball_test
usage: git stash list [<options>]
   or: git stash show [<stash>]
   or: git stash drop [-q|--quiet] [<stash>]
   or: git stash ( pop | apply ) [--index] [-q|--quiet] [<stash>]
   or: git stash branch <branchname> [<stash>]
   or: git stash [save [--patch] [-k|--[no-]keep-index] [-q|--quiet]
		       [-u|--include-untracked] [-a|--all] [<message>]]
   or: git stash clear
root@1c76e839f06b:/opt/repository# git version
git version 2.11.0
```

Beachball swallows the error and moves on to duplicate the CHANGELOG notes (it merges with main which brings the change files back and since git stash failed the first set of CHANGELOG changes are still there).

So long story short, changing to a newer container should work. The node:14.16-buster image has git 2.20 which should work fine.

Maybe this is worth documenting, or maybe not, since the world tends to move on from older versions. Feel free to close this issue, since it will remain findable by searching.

**Comment by ecraig12345** [comment]
Thanks for the investigation! I'm going to close this since like you mentioned, that version of git is quite old ([from 2017](https://github.com/git/git/releases/tag/v2.12.5)), so with limited resources to devote to beachball, it's not something we're likely to fix.
